gdal/ogr/ogrutils.cpp

```diff
@@ -64,6 +67,127 @@ inline bool CPLIsDoubleAnInt(double d)
     return d == static_cast<double>(static_cast<int>(d));
 }
 
+namespace
+{
+
+// Remove trailing zeros except the last one.
+std::string removeTrailingZeros(std::string s)
+{
+    auto pos = s.find('.');
+    if (pos == std::string::npos)
+        return s;
+
+    // Remove zeros at the end.  We know this won't be npos because we
+    // have a decimal point.
```

Comment:
why ? Can't s be "1.2345678" ?

Reply:
I don't understand.  find_last_not_of() will find '8' in this case, and there will be no truncation.

The comment is saying that there will be SOMETHING that's not zero in the string because the bit above guarantees that there will be a decimal point in the string.

Reply:
ah sorry, I read find_last_of() instead of find_last_not_of()

Reply:
I thought *I* was missing something.  Thanks.